specs/protocol/derivation.md

```diff
@@ -1278,3 +1284,29 @@ follows:
 
 [extended-attributes]: exec-engine.md#extended-payloadattributesv1
 [Fee Vaults]: exec-engine.md#fee-vaults
+
+## On Future-Proof Transaction Log Derivation
+
+As described in [L1 Retrieval][#l1-retrieval], batcher transactions' types are required to be from a fixed allow-list.
+
+However, we want to allow deposit transactions and `SystemConfig` update events to get derived even from receipts of
+future transaction types, as long as the receipts can be decoded following a best-effort process:
+
+As long as a future transaction type follows the [EIP-2718][https://eips.ethereum.org/EIPS/eip-2718] specification,
+the type can be decoded as the first byte of an encoded transaction or receipt (excluding legacy transactions, which are
+not _future_ transactions). We can then proceed as follows to get the logs of such a future transaction, or discard the
+transaction's receipt as invalid.
+
+- If it's a known transaction type, that is, legacy (first byte of the encoding is in the range `[0xc0, 0xfe]`) or its
+first byte is in the range `[0, 4]`, then it's not a _future transaction_ and we know how to decode the receipt and this
+process is irrelevant.
+- If a transaction's first byte is in the range `[5, 0x7f]`, it is expected to be a _future_ EIP-2718 transaction, so we can
+proceed to the receipt.
+- The receipt encoding's first byte must be the same byte as the transaction encoding's first byte, or it is discarded
+as invalid.
```

Comment:
```suggestion
- A non-legacy receipt encoding's first byte must be the same byte as the transaction encoding's first byte, or it is discarded
as invalid.
```

See code suggestion. I don't believe this is the case for legacy tx types. There it can be a range of bytes due to the RLP length. And tx length != receipt length.

Reply:
You're right for legacy transactions, but the process would already have stopped before reaching this step for legacy txs, see previous two steps. Will clarify nonetheless by adding `future`. Note that this is a description of a step that says, "if this is not given, then discard as invalid".